controllers/storagecluster/topology.go

```diff
@@ -135,7 +135,7 @@ func determinePlacementRack(
 	targetAZ := ""
 	for label, value := range node.Labels {
 		for _, key := range validTopologyLabelKeys {
-			if strings.Contains(label, key) && strings.Contains(label, "zone") {
+			if strings.Contains(label, key) && label == corev1.LabelZoneFailureDomainStable {
```

Comment:
For zone, we need to check `topology.kubernetes.io/zone` corev1.LabelTopologyZone
https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone

For host we need to check `kubernetes.io/hostname` corev1. LabelHostname
https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetesiohostname

For rack we need to check "topology.rook.io/rack"

Reply:
Why don't we check `hostname` and  `rack`  in the original code?

Reply:
That's due to the way failure domain are determined, We by default assume it to be rack, Then hostname if flexible scaling is true, if not at the end we check for zone.